src/main/groovy/edu/wpi/first/gradlerio/wpi/dependencies/tools/ToolInstallTask.groovy

```diff
@@ -57,7 +44,7 @@ class ToolInstallTask extends DefaultTask {
     }
 
     @CompileDynamic
-    static synchronized void setToolVersion(ToolJson tool) {
+    static synchronized void setToolVersion(WPIToolInfo tool) {

```

Comment:
What's the reasoning behind this being synchronized? Shouldn't two things being passed to it result in exactly the same thing being set both times so there's no benefits in blocking the other thread.

Reply:
This could be called from multiple tasks in parallel, which is modifying only part of a file on disk.

Reply:
Ah, I see. I suppose it is.
Why not just have a single task that writes to the tools file instead of having multiple tasks write to the tools file?

Reply:
Because there are multiple tools.

Reply:
Wouldn't it be better for each tool to generate it's own file and have whatever needs slurp them up. Then you don't have the multiple writer problem?

Reply:
No it wouldn't be. This file is defined as part of the VS Code setup, and we're just interfacing with that from gradle.